File: 艾诗缇/mdx_gyl/iosApiJson/ApidoAddYhkh.m
//
//  ApidoAddYhkh
//
//  Created by liulu on 2014-09-25 18:52:22
//  Copyright (c) liulu All rights reserved.


/**
   
*/

#import "ApidoAddYhkh.h"

@implementation ApidoAddYhkh


	/**
	 * 2.5	读取总数 useryhkh
	 * @param delegate 回调类
	 * @param select  回调函数
	 * @param class  分类 default com.shqj.webservice.entity.UserYHKH
	 * @param key  当前门店KEY
	 * @param khcity  当前门店KEY
	 * @param khsf  当前门店KEY
	 * @param psnname  当前门店KEY
	 * @param yhkh  当前门店KEY
	 * @callback MakeJFOrder
	*/
	-(UpdateOne*)get:(id)delegate selecter:(SEL)select  class:(NSString*)class key:(NSString*)key khcity:(NSString*)khcity khsf:(NSString*)khsf psnname:(NSString*)psnname yhkh:(NSString*)yhkh {
		NSMutableArray *array=[[NSMutableArray alloc]initWithObjects:nil];
		[array addObject:[NSString stringWithFormat:@"class=%@",class==nil?@"com.shqj.webservice.entity.UserYHKH":class==nil?@"":class]];
		[array addObject:[NSString stringWithFormat:@"key=%@",key==nil?@"":key]];
		[array addObject:[NSString stringWithFormat:@"khcity=%@",khcity==nil?@"":khcity]];
		[array addObject:[NSString stringWithFormat:@"khsf=%@",khsf==nil?@"":khsf]];
		[array addObject:[NSString stringWithFormat:@"psnname=%@",psnname==nil?@"":psnname]];
		[array addObject:[NSString stringWithFormat:@"yhkh=%@",yhkh==nil?@"":yhkh]];
		UpdateOne *updateone=[[Updateone2Json alloc] init:@"doAddYhkh" params:array  delegate:delegate selecter:select];
		return [self instanceUpdate:updateone];
	}

	/**
	 * 2.5	读取总数 useryhkh
	 * @param delegate 回调类
	 * @param select  回调函数
	 * @param class  分类 default com.shqj.webservice.entity.UserYHKH
	 * @param key  当前门店KEY
	 * @param khcity  当前门店KEY
	 * @param khsf  当前门店KEY
	 * @param psnname  当前门店KEY
	 * @param yhkh  当前门店KEY
	 * @callback MakeJFOrder
	*/
	-(UpdateOne*)load:(id)delegate selecter:(SEL)select  class:(NSString*)class key:(NSString*)key khcity:(NSString*)khcity khsf:(NSString*)khsf psnname:(NSString*)psnname yhkh:(NSString*)yhkh {
		UpdateOne *update=[self get:delegate selecter:select class:class key:key khcity:khcity khsf:khsf psnname:psnname yhkh:yhkh];
		[DataManager loadData:[[NSArray alloc]initWithObjects:update,nil] delegate:delegate];
		return update;
	}


@end

File: 艾诗缇/mdx_gyl/androidapis/com/mdx/store/data/proto/apis/ApidoAddYhkh.java
//
//  ApidoAddYhkh
//
//  Created by liulu on 2014-09-25 18:52:22
//  Copyright (c) liulu All rights reserved.


/**
   2.5	读取总数 useryhkh
*/

package com.mdx.store.data.proto.apis;

import android.content.Context;
import com.mdx.framework.server.api.ApiManager;
import com.mdx.framework.server.api.UpdateOne;


public class ApidoAddYhkh extends com.mdx.framework.server.api.ApiUpdate {


	/**
	 * 2.5	读取总数 useryhkh
	 * @param context 上下文
	 * @param parent  调用接口实体
	 * @param method  返回到哪个方法
	 * @param mclass 分类 default com.shqj.webservice.entity.UserYHKH
	 * @param key 当前门店KEY
	 * @param khcity 当前门店KEY
	 * @param khsf 当前门店KEY
	 * @param psnname 当前门店KEY
	 * @param yhkh 当前门店KEY
	 * @callback com.mdx.store.data.proto.GongT.MakeJFOrder
	*/
	public UpdateOne get(Context context, Object parent, String method,String mclass,String key,String khcity,String khsf,String psnname,String yhkh) {
		this.setMethod(method);
		this.setContext(context);
		this.setParent(parent);
		UpdateOne update = new UpdateOne("doAddYhkh", new String[][] { 
			{"class",mclass==null?"com.shqj.webservice.entity.UserYHKH":mclass},
			{"key",key},
			{"khcity",khcity},
			{"khsf",khsf},
			{"psnname",psnname},
			{"yhkh",yhkh}
		  });
		return initUpdateOne(update);
	}

	/**
	 * 2.5	读取总数 useryhkh
	 * @param context 上下文
	 * @param parent  调用接口实体
	 * @param method  返回到哪个方法
	 * @param mclass 分类 default com.shqj.webservice.entity.UserYHKH
	 * @param key 当前门店KEY
	 * @param khcity 当前门店KEY
	 * @param khsf 当前门店KEY
	 * @param psnname 当前门店KEY
	 * @param yhkh 当前门店KEY
	 * @callback com.mdx.store.data.proto.GongT.MakeJFOrder
	*/
	public UpdateOne load(Context context, Object parent, String method,String mclass,String key,String khcity,String khsf,String psnname,String yhkh) {
		UpdateOne update=get(context, parent, method,mclass,key,khcity,khsf,psnname,yhkh);
		ApiManager.Load(getContext(), getParent(), new UpdateOne[] { update });
		return update;

	}

	/**
	 * 2.5	读取总数 useryhkh
	 * @param context 上下文
	 * @param parent  调用接口实体
	 * @param method  返回到哪个方法
	 * @param mclass 分类 default com.shqj.webservice.entity.UserYHKH
	 * @param key 当前门店KEY
	 * @param khcity 当前门店KEY
	 * @param khsf 当前门店KEY
	 * @param psnname 当前门店KEY
	 * @param yhkh 当前门店KEY
	 * @callback com.mdx.store.data.proto.GongT.MakeJFOrder
	*/
	public ApidoAddYhkh set(String mclass,String key,String khcity,String khsf,String psnname,String yhkh) {
		get(null,null,null,mclass,key,khcity,khsf,psnname,yhkh);
		return this;
	}



	/**
	 * 2.5	读取总数 useryhkh
	 * @param context 上下文
	 * @param parent  调用接口实体
	 * @param method  返回到哪个方法
	 * @callback com.mdx.store.data.proto.GongT.MakeJFOrder
	*/
	public UpdateOne get(Context context, Object parent, String method) {
		this.setMethod(method);
		this.setContext(context);
		this.setParent(parent);
		UpdateOne update = new UpdateOne("doAddYhkh", new String[][] { 
			{"class","com.shqj.webservice.entity.UserYHKH"}
		 });
		return initUpdateOne(update);
	}

	/**
	 * 2.5	读取总数 useryhkh
	 * @param context 上下文
	 * @param parent  调用接口实体
	 * @param method  返回到哪个方法
	 * @callback com.mdx.store.data.proto.GongT.MakeJFOrder
	*/
	public UpdateOne load(Context context, Object parent, String method) {
		UpdateOne update=get(context, parent, method);
		ApiManager.Load(getContext(), getParent(), new UpdateOne[] { update });
		return update;

	}

	/**
	 * 2.5	读取总数 useryhkh
	 * @param context 上下文
	 * @param parent  调用接口实体
	 * @param method  返回到哪个方法
	 * @callback com.mdx.store.data.proto.GongT.MakeJFOrder
	*/
	public ApidoAddYhkh set() {
		get(null,null,null);
		return this;
	}


}
